weaviate/agents/__init__.pyi

```diff
@@ -0,0 +1,6 @@
+from weaviate.exceptions import WeaviateAgentsNotInstalledError
```

Comment:
do we need these .pyi files in addition to the .py file?

Reply:
Yes the `.pyi` stubs give us the ability to provide type hints for the optional package functionality when required by giving the type checkers the right hints but because they are not real `__init__.py` files the none of the paths will work at runtime if extras is not installed.